File: src/main/java/avalon/main/MainServlet.java
package avalon.main;

import avalon.friend.MainFriendMessageHandler;
import avalon.group.MainGroupMessageHandler;
import avalon.util.FriendMessage;
import avalon.util.GroupMessage;
import org.json.JSONObject;
import org.json.JSONTokener;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * Created by Eldath on 2017/1/28 0028.
 *
 * @author Eldath
 */
public class MainServlet extends HttpServlet {
    private static Logger logger = LoggerFactory.getLogger(MainServlet.class);

    @Override
    public void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        req.setCharacterEncoding("UTF-8");
        resp.setCharacterEncoding("UTF-8");
        JSONObject object = (JSONObject) new JSONTokener(req.getReader()).nextValue();
        if (object.isNull("post_type") || object.isNull("type")) return;
        if (!"receive_message".equals(object.getString("post_type")))
            return;
        //
        long timeLong = object.getLong("time");
        int Id = object.getInt("id");
        long senderUid;
        if (object.isNull("sender_uid")) {
            logger.error("\"sender_uid\" is null! \n " +
                    "That means the your Mojo-Webqq is outdated or " +
                    "Tencent delete some API. If this problem still exist, " +
                    "try update Mojo-Webqq or new issues at:" +
                    " https://github.com/sjdy521/Mojo-Webqq/issues.");
            senderUid = 1234567890;
        } else senderUid = object.getLong("sender_uid");
        String sender = object.get("sender").toString();
        String content = object.get("content").toString();
        String type = object.getString("type");
        if (!("friend_message".equals(type) || "group_message".equals(type)))
            return;
        if ("friend_message".equals(type)) {
            FriendMessage message = new FriendMessage(Id, timeLong, senderUid, sender, content);
            if (!MessageChecker.check(message)) return;
            MainFriendMessageHandler.getInstance().handle(message);
        } else {
            long groupUid;
            if (object.isNull("group_uid")) {
                logger.error("\"group_uid\" is null! \n " +
                        "That means the your Mojo-Webqq is outdated or " +
                        "Tencent delete some API. If this problem still exist, " +
                        "try update Mojo-Webqq or new issues at:" +
                        " https://github.com/sjdy521/Mojo-Webqq/issues.");
                groupUid = 1234567891;
            } else groupUid = object.getLong("sender_uid");
            String group = object.get("group").toString();
            MainGroupMessageHandler.getInstance().handle(new GroupMessage(Id, timeLong,
                    senderUid, sender, groupUid, group, content));
        }
    }
}

File: src/main/kotlin/avalon/main/MainServlet.kt
package avalon.main

import avalon.friend.FriendMessageHandler
import avalon.group.GroupMessageHandler
import avalon.util.FriendMessage
import avalon.util.GroupMessage
import org.json.JSONObject
import org.json.JSONTokener
import org.slf4j.LoggerFactory
import javax.servlet.http.*

/**
 * Created by Eldath on 2017/1/28 0028.
 *
 * @author Eldath
 */
@Deprecated("replaced by {@link avalon.util.servlet.AvalonServlet#setGroupMessageReceivedHook(Consumer)}")
class MainServlet : HttpServlet() {

	public override fun doPost(req: HttpServletRequest, resp: HttpServletResponse) {
		req.characterEncoding = "UTF-8"
		resp.characterEncoding = "UTF-8"
		val obj = JSONTokener(req.reader).nextValue() as JSONObject
		if (obj.isNull("post_type") || obj.isNull("type")) return
		if ("receive_message" != obj.getString("post_type"))
			return
		//
		val timeLong = obj.getLong("time")
		val id = obj.getInt("id")
		val senderUid = if (obj.isNull("sender_uid")) {
			logger.error("\"sender_uid\" is null! \n " +
					"That means the your Mojo-Webqq is outdated or " +
					"Tencent delete some API. If this problem still exist, " +
					"try update Mojo-Webqq or new issues at:" +
					" https://github.com/sjdy521/Mojo-Webqq/issues.")
			1234567890
		} else
			obj.getLong("sender_uid")
		val sender = obj["sender"].toString()
		val content = obj["content"].toString()
		val type = obj.getString("type")
		if (!("friend_message" == type || "group_message" == type))
			return
		if ("friend_message" == type) {
			val message = FriendMessage(id, timeLong, senderUid, sender, content)
			if (!MessageChecker.check(message)) return
			FriendMessageHandler.handle(message)
		} else {
			val groupUid = if (obj.isNull("group_uid")) {
				logger.error("\"group_uid\" is null! \n " +
						"That means the your Mojo-Webqq is outdated or " +
						"Tencent delete some API. If this problem still exist, " +
						"try update Mojo-Webqq or new issues at:" +
						" https://github.com/sjdy521/Mojo-Webqq/issues.")
				1234567891L
			} else
				obj.getLong("sender_uid")
			val group = obj["group"].toString()
			GroupMessageHandler.getInstance().handle(GroupMessage(id, timeLong,
					senderUid, sender, groupUid, group, content))
		}
	}

	companion object {
		private val logger = LoggerFactory.getLogger(MainServlet::class.java)
	}
}